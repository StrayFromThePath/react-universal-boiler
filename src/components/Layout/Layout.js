import React from 'react';
import Link from 'react-router-dom/Link';
import Helmet from 'react-helmet';
import PropTypes from 'prop-types';

const Layout = props =>
  <div>
    <Helmet titleTemplate="React Universal Boiler - %s">
      <html lang="en" />
      <title>React Universal Boiler</title>
      <meta name="application-name" content="React Universal Boiler" />
      <meta name="description" content="A server rendering React project." />
      <meta charSet="utf-8" />
      <meta httpEquiv="X-UA-Compatible" content="IE=edge" />
      <meta name="viewport" content="width=device-width, initial-scale=1" />
      <meta name="msapplication-TileColor" content="#2b2b2b" />
    </Helmet>
    <Link to="/tools">Tools</Link>
    <section>
      {props.children}
    </section>
  </div>;

const propTypes = {
  children: PropTypes.object,
};

Layout.propTypes = propTypes;

export default Layout;
