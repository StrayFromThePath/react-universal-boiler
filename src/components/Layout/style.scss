
.navbar {
  background-color: #23252C;
  color: #fff;
  width: 100%;
  height: 45px;
  display: flex;
  align-items: center;
  margin-bottom: 1rem;
  border-bottom: 1px solid darken(#23252C, 5%);
}

.wrapper {
  margin: 0 auto;
	max-width: 112.0rem;
	padding: 0 2.0rem;
	position: relative;
  width: 100%;
}
