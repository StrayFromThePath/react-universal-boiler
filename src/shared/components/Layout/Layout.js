import React from 'react';
import Link from 'react-router-dom/Link';
import PropTypes from 'prop-types';

const Layout = props => (
  <div>
    <nav className="navigation">
      <Link to="/tools">Tools</Link>
    </nav>
    <main className="container">{props.children}</main>
  </div>
);

Layout.propTypes = {
  children: PropTypes.object,
};

export default Layout;
