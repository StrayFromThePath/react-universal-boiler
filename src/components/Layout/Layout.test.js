import React from 'react';
import renderer from 'react-test-renderer';
import MemoryRouter from 'react-router-dom/MemoryRouter';
import Layout from './Layout';

jest.useFakeTimers();
Date.now = jest.fn(() => 1482363367071);

it('renders correctly', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Layout />
      </MemoryRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
