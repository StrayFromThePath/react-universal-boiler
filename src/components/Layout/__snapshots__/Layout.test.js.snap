// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <nav
    className="navbar"
  >
    <div
      className="wrapper"
    >
      <a
        href="/tools"
        onClick={[Function]}
      >
        Tools
      </a>
    </div>
  </nav>
  <main
    className="container"
  />
</div>
`;
